src/core/Akka.Persistence.Tests/FailedSnapshotStoreRecoverySpec.cs

```diff
@@ -0,0 +1,147 @@
+        protected override void OnCommand(object message)
+        {
+            switch (message)
+            {
+                case Save s:
+                {
+                    CurrentData = s.Data;
+                    SaveSnapshot(CurrentData);
+                    Sender.Tell("ack");
+                    break;
+                }
+                case Fetch:
+                {
+                    Sender.Tell(CurrentData);
+                    break;
+                }
+                case SaveSnapshotSuccess success:
+                {
+                    _log.Info("Snapshot saved");
+                    _targetActor.Tell(new SnapshotSaved());
+                    break;
+                }
+                case SaveSnapshotFailure failure:
+                {
+                    _log.Error(failure.Cause, "Snapshot failed");
+                    break;
+                }
+            }
+        }
+
+        protected override void OnRecover(object message)
+        {
+            switch (message)
+            {
+                case SnapshotOffer { Snapshot: string str }:
+                {
+                    CurrentData = str;
+                    break;
+                }
+            }
+        }
+
+        protected override void OnReplaySuccess()
+        {
+            _targetActor.Tell(new RecoveryCompleted());
+        }
+
+        protected override void OnRecoveryFailure(Exception reason, object message = null)
+        {
+            _log.Error(reason, "Recovery failed");
+            base.OnRecoveryFailure(reason, message);
+        }
+    }
+
+    [Theory(DisplayName = "PersistentActor using Snapshots only must fail if snapshots are irrecoverable")]
+    [InlineData(FailureMode.Explicit)]
+    [InlineData(FailureMode.Timeout)]
+    public async Task PersistentActor_using_Snapshots_only_must_fail_if_snapshots_irrecoverable(FailureMode mode)
+    {
+        // arrange
+        var probe = CreateTestProbe();
+        var actor = Sys.ActorOf(Props.Create(() => new PersistentActor("p1", probe.Ref)));
+        await probe.ExpectMsgAsync<RecoveryCompleted>();
+        actor.Tell(new Save("a"), probe);
+        await probe.ExpectMsgAsync("ack");
+        await probe.ExpectMsgAsync<SnapshotSaved>();
+        await actor.GracefulStop(RemainingOrDefault);
+
+        Task SelectBehavior(SnapshotStoreLoadBehavior behavior)
+        {
+            switch (mode)
+            {
+                case FailureMode.Timeout:
+                    return behavior.FailWithDelay(TimeSpan.FromMinutes(1));
+                case FailureMode.Explicit:
+                default:
+                    return behavior.Fail();
+            }
+        }
+
+        // act
+        await WithSnapshotLoad(SelectBehavior, async () =>
+        {
+            var actor2 = Sys.ActorOf(Props.Create(() => new PersistentActor("p1", probe.Ref)));
+            Watch(actor2);
+            await probe.ExpectNoMsgAsync();
```

Comment:
Should not receive an `SnapshotSaved` here as the actor will not successfully complete recovery under either scenario.